build/makelib/golang.mk

```diff
@@ -161,23 +156,15 @@ go.fmt:
 
 go.validate: go.vet go.fmt
 
-go.vendor: $(GLIDE) $(GLIDE_YAML)
-	@echo === updating vendor dependencies
-	@mkdir -p $(GLIDE_HOME)
-	@$(GLIDE) update --strip-vendor
+vendor: $(DEP)
+	@echo === ensure vendor dependencies
+	@$(DEP) ensure
 
-$(GLIDE_INSTALL_STAMP): $(GLIDE) $(GLIDE_LOCK)
-	@echo === installing vendor dependencies
-	@mkdir -p $(GLIDE_HOME)
-	@$(GLIDE) install --strip-vendor
-	@touch $@
-
-$(GLIDE):
-	@echo === installing glide
-	@mkdir -p $(TOOLS_HOST_DIR)/tmp
-	@curl -sL https://github.com/Masterminds/glide/releases/download/$(GLIDE_VERSION)/glide-$(GLIDE_VERSION)-$(GOHOSTOS)-$(GOHOSTARCH).tar.gz | tar -xz -C $(TOOLS_HOST_DIR)/tmp
-	@mv $(TOOLS_HOST_DIR)/tmp/$(GOHOSTOS)-$(GOHOSTARCH)/glide $(GLIDE)
-	@rm -fr $(TOOLS_HOST_DIR)/tmp
+$(DEP):
+	@echo === installing dep
+	@mkdir -p $(TOOLS_HOST_DIR)
+	@curl -sL -o $(DEP) https://github.com/golang/dep/releases/download/$(DEP_VERSION)/dep-$(GOHOSTOS)-$(GOHOSTARCH)
+	@chmod +x $(DEP)
```

Comment:
Generally, it is not a good idea to install publicly available software on user workstations without asking. Should the make validate if the tool installed and if not error out with a message for instructions on what is required?

Reply:
It’s installed in a private path. Our approach has been to install tools needed for the build in a private path that does not interfere with the host

Reply:
got it, thanks for the explanation.